src/Connector.js

```diff
@@ -252,19 +262,17 @@ class Connector {
     this.onReconnectedCb = _.noop();
 
     await createService(this.iotAgentUrl, this.orionUrl, '/device', 'default', 'device');
+    this.client = mqtt.connect(this.iotAgentMQTT);
+    await this.listen();
+  }
 
-    return new Promise((resolve, reject) => {
-      this.client = mqtt.connect(this.iotAgentMQTT);
-
-      this.client.on('connect', async () => {
-        const devices = await this.listDevices();
-        await subscribeToEntities(this.client, devices);
-        this.client.on('message', async (topic, payload) => {
-          await this.messageHandler(topic, payload);
-        });
-        return resolve();
+  async listen() {
+    await promisify(this.client, 'connect', async () => {
+      const devices = await this.listDevices();
```

Comment:
updated it! :+1:

Reply:
@ramonhpr, I attempted this approach in a couple of different ways and it didn't seem to work. I even tried adapting the `promisify` function to return `(...ret)` and to use `client.on` (instead of once) :disappointed: 

Considering this PR is mostly for improving code readability and testability (not a priority), it seemed reasonable to try other approaches that aren't as fancy, but just as readable and testable :innocent:

Let me know what you think about the current solution :rocket: